testsuite/tests/native-debugger/macos-lldb-amd64.reference

```diff
@@ -9,8 +9,8 @@ Executing commands in 'XXXX'.
 Breakpoint created for regex caml_start_program.
 (lldb) create caml_program
 Breakpoint created for regex caml_program.
-(lldb) create camlMeander\$c_to_ocaml*
-Breakpoint created for regex camlMeander\$c_to_ocaml*.
+(lldb) create camlMeander[.$]c_to_ocaml*
+Breakpoint created for regex camlMeander[.$]c_to_ocaml*.
```

Comment:
This change looks wrong.  Keep using `\$` ?

Reply:
Using the [.$] regexp allows to keep the same lldb script on all platforms. Otherwise, we would need to have a `lldb-linux-script` and a `lldb-script`. The quirky regexp seemed better to me than the duplication.

Reply:
The tests are sharing the same debugger [script](https://github.com/Octachron/ocaml/blob/stable_linux_mangling/testsuite/tests/native-debugger/lldb-script):
```bash
   debugger_script = "${test_source_directory}/lldb-script";
```

and it is this script which is echoed inside the reference file.